reconcile/utils/terrascript_client.py

```diff
@@ -3155,6 +3175,152 @@ def is_elasticsearch_domain_name_valid(name):
+        you should create and reuse a broader policy that includes multiple
+        log groups to avoid reaching this limit.
+        I.e., ideally we aggregate ALL log group identifiers for each
+        account first.
+        '''
+        log_group_infos = \
+            self._elasticsearch_get_all_log_group_infos()
+
+        log_groups_policy = {
+            'Version': '2012-10-17',
+            'Statement': [{
+                'Effect': 'Allow',
+                'Principal': {
+                    'Service': 'es.amazonaws.com',
+                },
+                'Action': [
+                    'logs:PutLogEvents',
+                    'logs:CreateLogStream',
+                ],
+                'Resource': [
+                    (
+                        f'arn:aws:logs:{info.region}:{info.account_id}'
+                        f':log-group:{info.log_group_identifier}:*'
+                    )
+                    for info in log_group_infos if info.account == account
+                ],
+            }]
+        }
+        log_groups_policy_values = {
+            'policy_name': 'es-log-publishing-permissions',
+            'policy_document': json.dumps(log_groups_policy, sort_keys=True),
+        }
+        resource_policy = aws_cloudwatch_log_resource_policy(
+            'es_log_publishing_resource_policy',
+            **log_groups_policy_values,
+        )
+        return resource_policy
+
+    def _get_tf_resource_elasticsearch_log_groups(
+        self, identifier: str, account: str,
+        resource: Mapping[str, Any], values: Mapping[str, Any],
+        output_prefix: str
+    ) -> tuple[list[dict[str, Any]], list[dict[str, str]]]:
+        ES_LOG_GROUP_RETENTION_DAYS = 180
+        tf_resources = []
+        publishing_options = []
+
+        # res.get('', []) won't work, as publish_log_types is
+        # explicitly set to None if not set
+        log_types = resource['publish_log_types']
+        if not log_types:
+            log_types = []
+        for log_type in log_types:
+            log_type_identifier = \
+                TerrascriptClient.elasticsearch_log_group_identifier(
+                    domain_identifier=identifier,
+                    log_type=ElasticSearchLogGroupType(log_type),
+                )
+            log_group_values = {
+                'name': log_type_identifier,
+                'tags': {},
+                'retention_in_days': ES_LOG_GROUP_RETENTION_DAYS,
+            }
+            region = values.get('region') or \
+                self.default_regions.get(account)
+            if self._multiregion_account(account):
+                log_group_values['provider'] = f'aws.{region}'
+            log_group_tf_resource = \
+                aws_cloudwatch_log_group(log_type_identifier,
+                                         **log_group_values)
+            tf_resources.append(log_group_tf_resource)
+            arn = f'${{aws_cloudwatch_log_group.{log_type_identifier}.arn}}'
```

Comment:
i believe you can use something simpler like `f'${{{aws_cloudwatch_log_group.arn}}}'`?

Reply:
Neat trick! Thx!